### ISSUE TITLE
Rendering raster styles when using OpenLayers MapboxVector layer

### ISSUE THREAD

**Issue by scadergit** [openlayers/ol-mapbox-style#641]
I am attempting to create a MapboxVector layer, as one of many other layers in a larger application. (Using `olms` is not feasible, given the management of the layers by the application.) The vector styling works great, but this ignores raster style types in the style definition. This behavior makes sense, but considering if the application defined a OL layer group could `olms` be restricted to do whatever it needed to within the group?

**Comment by ahocevar** [comment]
Currently there is no way to achieve what you want directly. But it should not be too hard to make `applyStyle` work with `Tile` layers, in addition to `Vector` and `VectorTile` layers.

**Comment by scadergit** [comment]
Would that mean making the Tile layer url parameter optional in exchange for providing a style url parameter that the Tile layer can pull the tile url from, or would you expect that to be done up-front, and simply invoke the applyStyle if an additional style url parameter is passed?

**Comment by ahocevar** [comment]
Currently, when you call `applyStyle()` with a `VectorLayer` or `VectorTile` layer instance, ol-mapbox-style knows what to do with it. The same can be implemented for `TileLayer`. If the `TileLayer` has a source already, `applyStyle` will override its url or url loading function, same as for a `VectorTile` source. If it has no source yet, `applyStyle()` will create one.

**Comment by scadergit** [comment]
Thanks...reviewed a lot of the code today, but missed that part. I'll look it over again.

**Comment by ahocevar** [comment]
@scadergit Thanks for digging into this. Let me know if I can help anywhere.

**Comment by scadergit** [comment]
In that approach, the application would need to know up-front if a style definition included raster and/or vector layers, create separate layers for each, and call `applyStyle()` on each?

**Comment by ahocevar** [comment]
Correct. But I see your point now. It looks like what you are looking for are `layers` and `sources` options for `apply()`. Adding those would also be an acceptable solution. Those could be an array of layer or source ids, or a function that receives a Mapbox layer or source as argument and returns a boolean to indicate whether to include or not.

**Comment by scadergit** [comment]
What if 'apply()` was updated (or create `applyGroup()`?) to apply the new layers of the style definition to an OL group, and ignore the any map/view style definition configuration?

**Comment by ahocevar** [comment]
It should be possible to extend the API so `apply()` also accepts a layer group in as first argument. In that case, `mapbox-style` property would be set on the layer group, and the background would be set on the first layer instead of the map.

**Comment by ahocevar** [comment]
It should be possible to extend the API so `apply()` also accepts a layer group in as first argument. In that case, `mapbox-style` property would be set on the layer group, and the background would be set on the first layer instead of the map.

**Comment by scadergit** [comment]
I have a wip to review, can I get push permissions?

**Comment by ahocevar** [comment]
Just create a pull request.

**Comment by scadergit** [comment]
Apparently my brain checked out early for the weekend...

**Comment by scadergit** [comment]
Resolved in #648 

Thanks, @ahocevar for your help on this effort.
